hs_tools_resource/models.py

```diff
@@ -207,19 +207,29 @@ def has_all_required_elements(self):
     def get_required_missing_elements(self):  # show missing required meta
         missing_required_elements = super(ToolMetaData, self).get_required_missing_elements()
 
-        if not self.url_bases.all().first():
-            missing_required_elements.append('App-launching Url Pattern')
-        elif not self.url_bases.all().first().value:
-            missing_required_elements.append('App-launching Url Pattern')
-
-        if not self.supported_res_types.all().first():
-            missing_required_elements.append('Supported Resource Types')
-        elif not self.supported_res_types.all().first().supported_res_types.count() > 0:
-            missing_required_elements.append('Supported Resource Types')
-
-        if self.supported_sharing_status.all().first():
-            if not self.supported_sharing_status.all().first().sharing_status.count() > 0:
-                missing_required_elements.append('Supported Sharing Status')
+        # At least one of the two metadata must exist: Home Page URL or App-launching URL Pattern
+        if (not self.url_bases.all().first() or not self.url_bases.all().first().value) \
```

Comment:
The way it should be implemented is that setting an empty string should fail the update action. Instead the user should be provided an option to delete an existing element the way we do with creators/contributors. I assume the base_url element has only one attribute (value). If there is no data for the value attribute then what's the use of the element?